extension/js/api.js

```diff
@@ -0,0 +1,95 @@
+import $ from 'jquery';
+import endpoints from './api_endpoints.json';
+
+var api_url = (Jupyter.notebook_list || Jupyter.notebook).base_url + "api/packagemanager/";
+
+/*
+This function makes the API calls to the specified endpoint with the corresponding headers
+*/
+
+function api_call(endpoint, url, payload, success) {
+
+    var settings = {
+        url: api_url + endpoint.uri + url,
+        method: endpoint.method,
+        headers: {
+            "Content-Type": "application/json",
+            "cache-control": "no-cache",
+        },
+        processData: false,
+        success: success
+    };
+
+    // Send the payload only if it is not empty
+
+    if (!(Object.keys(payload).length === 0 && payload.constructor === Object)) {
+        settings.data = JSON.stringify(payload);
+    }
+
+    $.ajax(settings);
```

Comment:
Nice! Finally! :)
~~But you don't need to get the promise from AJAX?~~ (didn't see that you were passing the `success`. I'm really happy to see this finally cleaned)